Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -1122,6 +1124,15 @@ public void RemoveDevice(InputDevice device, bool keepOnListOfAvailableDevices =
             // Remove state monitors while device index is still valid.
             RemoveStateChangeMonitors(device);
 
+            device.removing = true;
+            if (!device.remote)
+                UnsafeUtility.MemClear(device.currentStatePtr, device.valueSizeInBytes);
```

Comment:
To reset to defaults will need to copy from `defaultStatePtr`. For context, unfortunately, default state don't need to be zeroes. E.g. an 8-bit stick axis on a HID may have its 0 point at 127.

However, generally state shouldn't be mutated this way. We're basically changing inputs here in a way that bypasses the logic responsible for handling state changes.

I assume the goal is to get the action cancellations to return default values. I'm thinking that part is probably better handled in the action code instead of here as it affects cancellations in general and not just on device removal.

The fact that 'canceled' may not see a default value returned has turned out to be a problem in general. I'm guessing best thing is probably we just change that and make it always return `default(TValue)`.

Reply:
Hmm, you mean to have context.ReadValue always return default(TValue) for canceled actions? Yes, we can do that - that  would then be much closer to my original fix (which would return default if the device was removed), and would make all the RemoveDevice changes in this change unneeded.

Reply:
>you mean to have context.ReadValue always return default(TValue) for canceled actions?

Yeah, I'm starting to think that should be the case.

Not sure whether it'd obsolete the `removing` change outright. One thing that's great about it is that it now runs the action cleanup while the system setup for the device is still intact. So there won't be a case where, for example, a callback may find `InputAction.controls` to suddenly have controls that shouldn't be there.

But maybe I'm too concerned there about a detail that in practice won't really matter.